needy/library.py

```diff
@@ -105,6 +105,18 @@ def initialize_source(self):
         with OverrideEnvironment(self.__environment_overrides()):
             self.__post_clean()
 
+    def synchronize_source(self):
+        if 'download' in self.__configuration:
+            source = Download(self.__configuration['download'], self.__configuration['checksum'], self.source_directory(), os.path.join(self.directory(), 'download'))
+        elif 'repository' in self.__configuration:
+            source = GitRepository(self.__configuration['repository'], self.__configuration['commit'], self.source_directory())
+        elif 'directory' in self.__configuration:
+            source = Directory(self.__configuration['directory'] if os.path.isabs(self.__configuration['directory']) else os.path.join(self.needy.path(), self.__configuration['directory']), self.source_directory())
+        else:
+            raise ValueError('no source specified in configuration')
```

Comment:
hmmmmmmmm... these 9 lines look vaguely familiar...

i think you know what needs to be done here.